src/sentry/tasks/assemble.py

```diff
@@ -179,26 +189,38 @@ def get_assemble_status(task, scope, checksum):
     notice or error message.
     """
     cache_key = _get_cache_key(task, scope, checksum)
-    rv = default_cache.get(cache_key)
+
+    if options.get("assemble.read_from_redis"):
+        client = _get_redis_cluster_for_assemble()
+        rv = client.get(cache_key)
```

Comment:
Will this deserialize the tuple that was string cast?

Reply:
Yeah, fixed it. Thanks!